File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int h = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        
        if(m<45) {
            h=h-1;
            m=m+15;
        }
        else {
            m=m-45;
        }
        if(h==-1) {
            h=23;
        }
        System.out.print(h+" "+m);
    }
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.kt
import java.io.*
import java.util.*

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())

    val h = Integer.parseInt(st.nextToken())
    val m = Integer.parseInt(st.nextToken())

    setAlarm(h, m)
}

fun setAlarm(h: Int, m: Int) {
    var minute = m;
    var hour = h;
    if (minute < 45) {
        minute +=15
        hour-=1
    } else minute -= 45

    if(hour==-1) hour = 23

    println("$hour $minute")
}
